File: mestrado/estrutura_dados/aula03/fila.py
def inserirFila(novo_valor, fila, topo, tamanho):
    if topo == tamanho:
        print("Fila cheia!")
    else:
        fila[topo] = novo_valor
        topo += 1

def removerFila(fila, topo, tamanho):
    if topo == 0:
        print("Fila vazia!")
        return -1
    else:
        valor = fila[topo - 1]
        topo -= 1
        return valor
    
def removerFila2(F, f, r):
    if f != 0:
        valor = F[f]
        if f == r:
            r = 0
        else:
            f = (f % r) + 1
        return valor
    else:
        print("Fila vazia!")

def inserirFila2(F, f, r, M, novo_valor):
    prox = (r % M) + 1
    if prox != f:
        r = prox
        F[r] = novo_valor
        if f == 0:
            f = 1
    else:
        print("Fila cheia!")

if __name__ == "__main__":
    fila = [0] * 10
    topo = 0
    valor = removerFila(fila, topo, 10)
    print("Valor removido:", valor)

    valor = removerFila(fila, topo, 10)
    print("Valor removido:", valor)

    valor = removerFila(fila, topo, 10)
    print("Valor removido:", valor)

    print(removerFila(fila, topo, 10))
    inserirFila(1, fila, topo, 10)
    inserirFila(2, fila, topo, 10)
    inserirFila(3, fila, topo, 10)
    inserirFila(4, fila, topo, 10)
    inserirFila(5, fila, topo, 10)
    print(removerFila2(fila, topo, 10))
    print(removerFila2(fila, topo, 10))

File: mestrado/estrutura_dados/aula03/fila.c
#include <stdio.h>

void inserirFila(int novo_valor, int fila[], int *topo, int tamanho) {
    if (*topo == tamanho) {
        printf("Fila cheia!\n");
    } else {
        fila[(*topo)++] = novo_valor;
    }
}

int removerFila(int fila[], int *topo, int tamanho) {
    if (*topo == 0) {
        printf("Fila vazia!\n");
        return -1;
    } else {
        return fila[--(*topo)];
    }
}


int main(void) {
    int fila[10];
    int topo = 0;
    int valor;

    inserirFila(1, fila, &topo, 10);
    inserirFila(2, fila, &topo, 10);
    inserirFila(3, fila, &topo, 10);
    inserirFila(4, fila, &topo, 10);
    inserirFila(5, fila, &topo, 10);

    valor = removerFila(fila, &topo, 10);
    printf("Valor removido: %d\n", valor);

    valor = removerFila(fila, &topo, 10);
    printf("Valor removido: %d\n", valor);

    valor = removerFila(fila, &topo, 10);
    printf("Valor removido: %d\n", valor);

    printf("Fila vazia: %d\n", removerFila(fila, &topo, 10));

    return 0;
}